apps/src/code-studio/pd/teacher_application/teacher_application.jsx

```diff
@@ -0,0 +1,374 @@
+          id="schoolEmail"
+          label="Your personal email address (we may need to contact you during the summer)"
+          type="email"
+        />
+        <FieldGroup
+          id="phoneNumber"
+          label="Preferred phone number"
+          type="text"
+        />
+        {this.buttonList('radio', 'Gender Identity', 'genderIdentity', ['Female', 'Male', 'Other', 'Prefer not to answer'])}
+        {this.buttonList('check', 'What grades are you teaching in the current 2016-17 school year? (select all that apply)', 'grades2016', grades)}
+        {this.buttonList('check', 'What subjects are you teaching in the current 2016-17 school year? (select all that apply)', 'subjects2016', subjects, true)}
+        {this.buttonList('check', 'What grades are you teaching in the current 2017-18 school year? (select all that apply)', 'grades2017', grades)}
+        {this.buttonList('check', 'What subjects are you teaching in the current 2016-17 school year? (select all that apply)', 'subjects2017', subjects, true)}
+        <FieldGroup
+          id="principalFirstName"
+          label="Principal's first name"
+          type="text"
+        />
+        <FieldGroup
+          id="principalLastName"
+          label="Principal's last name"
+          type="text"
+        />
+        <FormGroup>
+          <ControlLabel>
+            Principal's prefix
+          </ControlLabel>
+          <FormControl componentClass="select" id="principalPrefix">
+            <option value="Dr.">Dr.</option>
+            <option value="Miss.">Miss.</option>
+            <option value="Mr.">Mr.</option>
+            <option value="Mrs.">Mrs.</option>
+            <option value="Ms.">Ms.</option>
+          </FormControl>
+        </FormGroup>
+        <FieldGroup
+          id="principalEmail"
+          label="Principal's Email Address"
+          type="text"
+        />
+      </div>
+    );
+  },
+
+  renderCourseSelection() {
+    return (
+      <div>
+        <ControlLabel>
+          Which professional learning program are you applying to join for the 2017-18 school year? Click on each
+          curriculum for more information. Note: this application is only for Computer Science Discoveries and Computer
+          Science Principles. If you are interested in <a href="">Computer Science Fundamentals</a>, please
+          visit <a href="">this</a> page for information about workshops in your area. (Select one)
+        </ControlLabel>
+        <Radio
+          value="csd"
+          name="courseSelection"
+          onChange={this.handleCourseChange}
+        >
+          <a href="">Computer Science Discoveries</a> (designed for 7th - 9th grade)
+        </Radio>
+        <Radio
+          value="csp"
+          name="courseSelection"
+          onChange={this.handleCourseChange}
+        >
+          <a href="">Computer Science Principles</a> (designed for 9th - 12th grade, and can be implemented as an AP or introductory course)
+        </Radio>
```

Comment:
Is this only these 2? Certainly if there might be more, and even with just these it might be nicer to have the values in an array of objects and map to create the radios:

```jsx
const COURSES=[
  {
    value='csd',
    name='Computer Science Discoveries'
    href='...' // assuming there will actually be urls here?
    subtitle=' (designed for 7th - 9th grade)'
  }, {
    value='csp'
    // ...
  }

// ...

COURSES.map( course => (
  <Radio
    value={course.value}
    name="courseSelection"
    onChange={this.handleCourseChange}
  >
    <a href={course.href}>course.name</a>
    {course.subtitle}
  </Radio>
));
]
```